### PR TITLE
refactor: simplify converting List DataTypes to `ScalarValue`

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

N/A

## Rationale for this change
`ScalarValue::try_from(dt: &DataType)`  is commonly used to create a NULL `ScalarValue` for a given DataType.
The processing of List DataTypes is a bit complex and can be simplified.

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/common/src/scalar/mod.rs [hunk 1]
```diff
@@ -3089,46 +3089,21 @@ impl TryFrom<&DataType> for ScalarValue {
...
+            // `ScalarValue::LargeList` contains single element `LargeListArray`.
+            DataType::LargeList(field_ref) => ScalarValue::LargeList(Arc::new(
+                GenericListArray::new_null(field_ref.clone(), 1),
+            )),
             // `ScalaValue::FixedSizeList` contains single element `FixedSizeList`.
-            DataType::FixedSizeList(field, _) => ScalarValue::FixedSizeList(
-                new_null_array(
-                    &DataType::FixedSizeList(
-                        Arc::new(Field::new("item", field.data_type().clone(), true)),
-                        1,
```
Comment (jonahgao) [inline datafusion/common/src/scalar/mod.rs:3124]: Here, it should be the fixed length instead of always being `1`.

**Review by alamb — APPROVED** [review]
Thank you @jonahgao  -- this makes sense to me and I think makes the code look much nicer. 

@jayzhan211 if you have time can you also give this PR a review?

**Review by jayzhan211 — APPROVED** [review]
I want to point out that this change is different from the previous one. The current change creates an **empty array**, whereas the previous one created an **array with null**. Although I'm not certain which version is preferred, both satisfy the requirements: 1. the row length is one, as in ScalarValue, and 2. it is null. Therefore, I believe either option is acceptable.

**Comment by jonahgao** [comment]
Thank you for the reviewing @alamb @jayzhan211 ❤️

**Pull request merged**